test/distributions/block/arrayAssignment.chpl

```diff
@@ -0,0 +1,63 @@
+// test from https://github.com/chapel-lang/chapel/issues/18747
+module Spam {
+    use BlockDist;
+    use CTypes;
+    use IO;
+    use IO.FormattedIO;
+    use Reflection;
+
+    type region_desc = c_ptr(void);
+
+    class Sam {
+        proc lookup_region(regionName:string): region_desc throws {
+            return nil;
+        }
+    }
+
+    class Region {
+        const regionName: string;
+        var regionDescs = blockDist.createArray({0..#numLocales}, region_desc);
+        const size: uint(64);
+        var iAmDestroyed = false;
+
+        proc init(regionName: string, size: uint(64)) {
+            this.regionName = regionName;
+            this.size = size;
+        }
+
+    }
+    proc newSharedRegion(regionName: string, size: uint(64),
+                         ref regionDescs: [] region_desc) throws {
+        try! stderr.writeln("%s %i".format(getRoutineName(), getLineNumber()));
+        var region = new shared Region(regionName, size);
+        try! stderr.writeln("%s %i".format(getRoutineName(), getLineNumber()));
+        region.regionDescs = regionDescs;
```

Comment:
Definitely good to use this line, but I might be inclined to add a version of the test with the part John had commented out, on the same argument as the Greg issue :)